geonode/layers/data.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+#########################################################################
+#
+# Copyright (C) 2014 OpenPlans
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#########################################################################
+
+"""Utilities for accessing data linked to GeoNode layers
+"""
+
+# Standard Modules
+import csv
+import logging
+
+# Django functionality
+from django.conf import settings
+from django.http import HttpResponse
+
+# Imports for SOS
+if 'geonode.geoserver' in settings.INSTALLED_APPS:
+    from geonode.geoserver.ows import sos_swe_data_list, sos_observation_xml
+
+logger = logging.getLogger("geonode.layers.data")
+
+
+def layer_sos(feature, supplementary_info={}, time=None, mimetype="text/csv"):
+    """Return SOS data in mimetype format for a layer that specifies a valid URL.
+
+    Parameters
+    ----------
+    feature : string
+        the ID of a feature from the WFS; this is used as a link to a
+        corresponding "feature_of_interest" in the SOS
+    supplementary_info : dictionary
+        set of parameters used to specify access to the SOS.  For example:
+            {"sos_url": "http://sos.server.com:8080/sos",
+             "observedProperties": ["urn:ogc:def:phenomenon:OGC:1.0.30:temperature"],
+             "offerings": ["WEATHER"]}
+    time : string
+        Optional.   Time should conform to ISO format: YYYY-MM-DDTHH:mm:ss+-HH
+        Instance is given as one time value. Periods of time (start and end) are
+        separated by "/". Example: 2009-06-26T10:00:00+01/2009-06-26T11:00:00+01
+
+    Returns
+    ------
+    HttpResponse with mimetype data, or None if an error is encountered.
+    """
+    sos_data = HttpResponse(content="Unable to process request", status=500)
+    offerings = supplementary_info.get('offerings')
+    url = sup_supplementary_info.get('sos_url')
```

Comment:
This variable does not exist. It should probably be supplementary_info and this hints that the code may not be working properly.

Unfortunately we do not have an SOS server to test against and will have to pass on this.